src/main/java/com/google/devtools/build/lib/bazel/repository/downloader/Checksum.java

```diff
@@ -31,6 +31,12 @@ private InvalidChecksumException(String msg) {
     }
   }
 
+  public static final class ICExceptionExtended extends Exception {
+    private ICExceptionExtended(KeyType keyType, String hash, String correctHash) {
+      super("Invalid " + keyType + " checksum '" + hash + "'. The correct checksum is '" + correctHash + "'.");
```

Comment:
nit: line length, and also I think that exception messages generally don't have trailing periods.